src/components/NewEventDialog.js

```diff
@@ -0,0 +1,156 @@
+import React, { useCallback } from 'react';
+import { useMutation } from '@apollo/react-hooks';
+import useForm from 'react-hook-form';
+import Dialog from '@material-ui/core/Dialog';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import { TextField, Button, Grid, Box } from '@material-ui/core';
+import PropTypes from 'prop-types';
+import gql from 'graphql-tag';
+
+import useStyle from './UseStyle';
+
+const mutation = gql`
+  mutation addEvent(
+    $title: String!
+    $start: String!
+    $end: String!
+    $description: String!
+  ) {
+    addEvent(
+      title: $title
+      start: $start
+      end: $end
+      description: $description
+    ) {
+      id
+      title
+      start
+      end
+      description
+    }
+  }
+`;
+
+function NewEventDialog(props) {
+  const { onClose, open, selectedDate } = props;
+  const classes = useStyle();
+
+  const handleClose = () => {
+    onClose();
+  };
+  const { handleSubmit, register, errors } = useForm();
+  const [addEvent, { loading, error }] = useMutation(mutation);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const onSubmit = useCallback(
+    ({ title, description, date, startTime, endTime }) => [
+      addEvent({
+        variables: {
+          title,
+          start: startTime ? date + 'T' + startTime : date,
```

Comment:
Can `startTime` be undefined if it's `required` in the component?

Reply:
This format `yyyy-MM-ddThh:mm` appears to be the format of [datetime-local](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local), which is of course not very useful when dealing with timezones, and perhaps the stored events should be proper ISO strings, so that they are useful even if switching the timezone.